core/src/test/java/com/ericsson/bss/cassandra/ecchronos/core/TestCASLockFactory.java

```diff
@@ -375,13 +380,103 @@ public boolean getRemoteRouting()
                             {
                                 return true;
                             }
+
+                            @Override
+                            public String getSerialConsistency(){
+                                return "DEFAULT";
+                            }
                         })
                         .withHostStates(hostStates)
                         .withStatementDecorator(s -> s)
                         .withKeyspaceName(myKeyspaceName)
                         .build());
     }
 
+    @Test
+    public void testRemoteRoutingTrueWithDefaultSerialConsistency() {
+
+        myLockFactory = new CASLockFactory.Builder()
+                .withNativeConnectionProvider(getNativeConnectionProvider())
+                .withHostStates(hostStates)
+                .withStatementDecorator(s -> s)
+                .withKeyspaceName(myKeyspaceName)
+                .build();
+
+        
+        assertEquals(ConsistencyLevel.LOCAL_SERIAL, myLockFactory.getSerialConsistencyLevel());
+    }
+
+    @Test
+    public void testRemoteRoutingFalseWithDefaultSerialConsistency() {
+
+        Node nodeMock = mock(Node.class);
+        
+        NativeConnectionProvider connectionProviderMock = mock(NativeConnectionProvider.class);
+
+        when(connectionProviderMock.getSession()).thenReturn(mySession);
+
+        when(connectionProviderMock.getLocalNode()).thenReturn(nodeMock);
+        
+        when(connectionProviderMock.getRemoteRouting()).thenReturn(false);
+
+        when(connectionProviderMock.getSerialConsistency()).thenReturn("DEFAULT");
+
+        myLockFactory = new CASLockFactory.Builder()
+                .withNativeConnectionProvider(connectionProviderMock)
+                .withHostStates(hostStates)
+                .withStatementDecorator(s -> s)
+                .withKeyspaceName(myKeyspaceName)
+                .build();
+
+        assertEquals(ConsistencyLevel.SERIAL, myLockFactory.getSerialConsistencyLevel());
+    }
+
+    @Test
+    public void testLocalSerialConsistency(){
+
+        NativeConnectionProvider connectionProviderMock = mock(NativeConnectionProvider.class);
+
+        Node nodeMock = mock(Node.class);
+        
+        when(connectionProviderMock.getSerialConsistency()).thenReturn("LOCAL");
+
+        when(connectionProviderMock.getSession()).thenReturn(mySession);
+
+        when(connectionProviderMock.getLocalNode()).thenReturn(nodeMock);
+        
+        myLockFactory = new CASLockFactory.Builder()
+                .withNativeConnectionProvider(connectionProviderMock)
+                .withHostStates(hostStates)
+                .withStatementDecorator(s -> s)
+                .withKeyspaceName(myKeyspaceName)
+                .build();
+
+        assertEquals(ConsistencyLevel.LOCAL_SERIAL, myLockFactory.getSerialConsistencyLevel());
+
+    }
+
+    @Test
+    public void testSerialConsistency(){
```

Comment:
Incorrect formatting {} should be placed on newlines.

Reply:
Done